flybywiresim/a32nx
# feat: engine shutdown and start-up/ shutdown animation

Fixes #3169
Fixes #5119
Fixes #6301

## Summary of Changes

This is a dual PR:
(i) Implements realistic engine shutdown, taking into account primary parameter behavior, including realistic EGT cooldown, and engine delay between master off and spool-down (around 1.2 seconds)

(ii) Realistic engine animation behavior at start-up and shutdown. Animation is now attached to the FBW N1 LVar, with fan starting right at the beginning of the process (not after ignition).

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): TazX [Z+2]#0405

## Testing instructions

(1) Testing for Engine Shutdown:
- Start aircraft on the runway
- Shut one of the engines
- Engine spool-down should start between 1 and 2 seconds afterwards
- N1, N2 and FF should begin to go down non-linearly
- EGT should drop down fast until a certain point; then it will maintain heat energy and drop down very slowly until ambient temperature (it could take well over a number of hours)

(2) Testing for engine animation:
- Start the aircraft C&D
- Proceed to 'wake' the aircraft in order to start the engines
- Go to External view and set yourself facing the engine. Go back in to the cockpit
- To control engine spool while on outside view, decouple the E/WD
- Set Engine Mode to IGN and Start an Engine
- Go to the External View and watch the animation:
       (i) Engine should start spooling around 1 second after Master is on
       (ii) Engine rotation follows the start-up (and shutdown) FBW N1 (non-linear)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
